frontend/packages/ceph-storage-plugin/src/components/modals/storage-pool-modal/storage-pool-modal.tsx

```diff
@@ -0,0 +1,391 @@
+          <EmptyStateBody>Pool {name} was successfully created</EmptyStateBody>
+        </EmptyState>
+      )}
+      {status === 'failed' && (
+        <EmptyState>
+          <EmptyStateIcon icon={ExclamationCircleIcon} className="ceph-storage-pool__error-icon" />
+          <EmptyStateBody>
+            {error || `An error occurred, Pool ${name} was not created`}
+          </EmptyStateBody>
+        </EmptyState>
+      )}
+      {status === 'timeout' && (
+        <EmptyState>
+          <EmptyStateIcon icon={UnknownIcon} className="ceph-storage-pool__timeout-icon" />
+          <EmptyStateBody>
+            Pool creation timed out. Please check the events to know pool {name} status
+          </EmptyStateBody>
+        </EmptyState>
+      )}
+    </div>
+  );
+};
+
+export const StoragePoolModal = withHandlePromise((props: StoragePoolModalProps) => {
+  const {
+    cephClusterObj,
+    onPoolCreation,
+    close,
+    cancel,
+    handlePromise,
+    errorMessage,
+    inProgress,
+  } = props;
+  const [newPoolName, setNewPoolName] = React.useState('sc-pool');
+  const [isReplicaOpen, setReplicaOpen] = React.useState(false);
+  const [isPerfObjOpen, setPerfObjOpen] = React.useState(false);
+  const [replicaSize, setReplicaSize] = React.useState('');
+  const [isCompressed, setCompression] = React.useState(false);
+  const [deviceClass, setdeviceClass] = React.useState('');
+  const [poolStatus, setPoolStatus] = React.useState('');
+  const [storageClusterLoading, setStorageClusterLoading] = React.useState(true);
+  const [storageClusterError, setStorageClusterError] = React.useState(false);
+  const [storageClusterReady, setStorageClusterReady] = React.useState(false);
+  const [isSubmit, setIsSubmit] = React.useState(false);
+  const [timeoutClear, setTimeClear] = React.useState(false);
+  const [timer, setTimer] = React.useState<NodeJS.Timer>(null);
+
+  const poolResource = React.useMemo(() => {
+    return {
+      kind: referenceForModel(CephBlockPoolModel),
+      namespaced: true,
+      isList: false,
+      name: newPoolName,
+    };
+  }, [newPoolName]);
+
+  const [newPool, newPoolLoaded, newPoolLoadError] = useK8sWatchResource<StoragePoolKind>(
+    poolResource,
+  );
+
+  React.useEffect(() => {
+    k8sGet(OCSServiceModel, OCS_INTERNAL_CR_NAME, CEPH_STORAGE_NAMESPACE)
+      .then((cluster: OCSStorageClusterKind) => {
+        setStorageClusterLoading(false);
+        if (cluster.status.phase === 'Ready') {
+          setStorageClusterReady(true);
+        }
+      })
+      .catch((e) => setStorageClusterError(e));
+    if (timer && timeoutClear) {
+      clearTimeout(timer);
+    }
+    if (isSubmit) {
+      if (newPool && newPoolLoaded && newPool?.status?.phase === 'Ready') {
+        setPoolStatus('created');
+        setIsSubmit(false);
+        setTimeClear(true);
+      } else if (newPoolLoaded && newPool?.status?.phase === 'ReconcileFailed') {
```

Comment:
Create an enum for all possible phase values.

Reply:
ack